source/simulator/newton.cc

```diff
@@ -161,6 +161,239 @@ namespace aspect
+                           "before switching to Newton iterations.");
+
+        prm.declare_entry ("Max Newton line search iterations", "5",
+                           Patterns::Integer (0),
+                           "The maximum number of line search iterations allowed. If the "
+                           "criterion is not reached after this iteration, we apply the scaled "
+                           "increment to the solution and continue.");
+
+        prm.declare_entry ("Use Newton residual scaling method", "false",
+                           Patterns::Bool (),
+                           "This method allows to slowly introduce the derivatives based on the improvement "
+                           "of the residual. If set to false, the scaling factor for the Newton derivatives "
+                           "is set to one immediately when switching on the Newton solver.");
+
+        prm.declare_entry ("Maximum linear Stokes solver tolerance", "0.9",
+                           Patterns::Double (0,1),
+                           "When the linear Stokes solver tolerance is dynamically chosen, this defines "
+                           "the most loose tolerance allowed.");
+
+        prm.declare_entry ("Stabilization preconditioner", "SPD",
+                           Patterns::Selection ("SPD|PD|symmetric|none"),
+                           "This parameters allows for the stabilisation of the preconditioner. By default, the "
+                           "matrix created for the preconditioning is not necessarily Symmetric Positive Definite. "
+                           "This is problematic (see Fraters et al, in prep). When none is chosen, the perconitioner "
+                           "is not stabilized. The Symmetric parameters symmetrizes the matrix, and PD makes the matrix "
+                           "Positive Definite. SPD is the full stabilization, where the matrix is garanteed Symmetric "
+                           "Positive Definite.");
+        prm.declare_entry ("Stabilization velocity block", "SPD",
+                           Patterns::Selection ("SPD|PD|symmetric|none"),
+                           "This parameters allows for the stabilisation of the velocity block. By default, the "
+                           "matrix created for the velocity block is not necessarily Symmetric Positive Definite. "
+                           "This is problematic (see Fraters et al, in prep). When none is chosen, the velocity block "
+                           "is not stabilized. The Symmetric parameters symmetrizes the matrix, and PD makes the matrix "
+                           "Positive Definite. SPD is the full stabilization, where the matrix is garanteed Symmetric "
+                           "Positive Definite.");
+        prm.declare_entry ("Use Newton failsafe", "false",
+                           Patterns::Bool (),
+                           "Switches on SPD stabilization when solver fails.");
```

Comment:
Can you elaborate on this in a couple more sentences? I.e., *what* solver needs to fail, and maybe reference what other parameter values are then changed? And for how long -- all future time steps, or just the current time step?